### PR TITLE
19 alerting concept

### PR DESCRIPTION
**Pull request by Elwizzy12**
The pupose of this ticket creating an  alerting concept for the mobile banking application. 
The alerting system is to enhance user engagement, provide timely notifications about important transactions, and ensure the security of user accounts.

### CODE REVIEW

File: docs/concepts/alerting.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+3. **Condition Check**: The system checks if the transaction meets trigger conditions.
+4. **Alert Generation**: An alert is created and queued for delivery.
+5. **Notification Delivery**: The alert is sent to the user via their preferred channel.
+6. **User Interface Update**: The alert appears in the user’s alert dashboard.
+---
+## 11. Conclusion
+
+The alerting concept for the online banking application is designed to enhance user engagement and security through timely notifications. By implementing customizable alert preferences and ensuring efficient alert management, the application aims to provide a seamless and secure banking experience for users.
+
+By integrating Prometheus for monitoring and Grafana for visualization, the alerting system will be robust and provide valuable insights into application performance and user activity.
```
Comment (Koufan-De-King) [inline docs/concepts/alerting.md:122]: I don't believe a conclusion is necessary @Elwizzy12

**Review by Koufan-De-King — CHANGES_REQUESTED** [review]

File: docs/concepts/alerting.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+### 3.1 Transaction Alerts
+- **Description**: Notifications for account transactions such as deposits, withdrawals, and transfers.
+- **Trigger Conditions**: Any transaction exceeding a predefined amount or any transaction made by the user.
+
+### 3.2 Balance Alerts
+- **Description**: Notifications when the account balance falls below a specified threshold.
+- **Trigger Conditions**: Balance drops below the user-defined limit.
+
+### 3.3 Security Alerts
```
Comment (Blindspot22) [inline docs/concepts/alerting.md:27]: I think you should also include `Alert` for maybe more than 5 or 10 failed login attempts?
What do you think @Elwizzy12  or is that what `unauthorized login attempts` in your description mean?

**Review by Blindspot22 — CHANGES_REQUESTED** [review]

File: docs/concepts/alerting.md
Comment (francis-pouatcha) [inline docs/concepts/alerting.md:1]: I believe there's a misunderstanding between technical alerts and business notifications.

- Initially, we were referring to technical alerts related to the application's state.
- However, business notifications are also something we would eventually need to address.

Please explain the concept of technical alerts here, and move the work related to business notifications into a separate file called **"Requirements for User Notifications."**

**Review by francis-pouatcha — CHANGES_REQUESTED** [review]

**Review by Koufan-De-King — APPROVED** [review]

**Review by nancymuyeh — APPROVED** [review]

**Review by francis-pouatcha — APPROVED** [review]

**Review by Blindspot22 — APPROVED** [review]
Looks Good to Me :100:

**Pull request merged**